source/include/azure_iot_hub_client.h

```diff
@@ -0,0 +1,428 @@
+
+    /* Service success codes */
+    eAzureIoTStatusOk = 200,
+    eAzureIoTStatusAccepted = 202,
+    eAzureIoTStatusNoContent = 204,
+
+    /* Service error codes */
+    eAzureIoTStatusBadRequest = 400,
+    eAzureIoTStatusUnauthorized = 401,
+    eAzureIoTStatusForbidden = 403,
+    eAzureIoTStatusNotFound = 404,
+    eAzureIoTStatusNotAllowed = 405,
+    eAzureIoTStatusNotConflict = 409,
+    eAzureIoTStatusPreconditionFailed = 412,
+    eAzureIoTStatusRequestTooLarge = 413,
+    eAzureIoTStatusUnsupportedType = 415,
+    eAzureIoTStatusThrottled = 429,
+    eAzureIoTStatusClientClosed = 499,
+    eAzureIoTStatusServerError = 500,
+    eAzureIoTStatusBadGateway = 502,
+    eAzureIoTStatusServiceUnavailable = 503,
+    eAzureIoTStatusTimeout = 504,
+} AzureIoTHubMessageStatus_t;
+
+/**
+ * @brief IoT Hub Cloud Message Request struct.
+ */
+typedef struct AzureIoTHubClientCloudToDeviceMessageRequest
+{
+    const void * pvMessagePayload;           /**< The pointer to the message payload. */
+    uint32_t ulPayloadLength;                /**< The length of the message payload. */
+
+    AzureIoTMessageProperties_t xProperties; /**< The bag of properties received with the message. */
+} AzureIoTHubClientCloudToDeviceMessageRequest_t;
+
+/**
+ * @brief IoT Hub Method Request struct.
+ */
+typedef struct AzureIoTHubClientMethodRequest
+{
+    const void * pvMessagePayload; /**< The pointer to the message payload. */
+    uint32_t ulPayloadLength;      /**< The length of the message payload. */
+
+    const uint8_t * pucRequestID;  /**< The pointer to the request ID. */
+    int16_t usRequestIDLength;     /**< The length of the request ID. */
+
+    const uint8_t * pucMethodName; /**< The name of the method to invoke. */
+    uint16_t usMethodNameLength;   /**< The length of the method name. */
+} AzureIoTHubClientMethodRequest_t;
+
+/**
+ * @brief IoT Hub Twin Response struct.
+ */
+typedef struct AzureIoTHubClientTwinResponse
+{
+    AzureIoTHubMessageType_t messageType;     /**< The type of message received. */
+
+    const void * pvMessagePayload;            /**< The pointer to the message payload. */
+    uint32_t ulPayloadLength;                 /**< The length of the message payload. */
+
+    uint32_t ulRequestID;                     /**< The request ID for the twin response. */
+
+    AzureIoTHubMessageStatus_t messageStatus; /**< The operation status. */
+
+    const uint8_t * pucVersion;               /**< The pointer to the twin document version. */
+    uint16_t usVersionLength;                 /**< The length of the twin document version. */
+} AzureIoTHubClientTwinResponse_t;
+
+/**
+ * @brief Cloud message callback to be invoked when a cloud message is received in the call to AzureIoTHubClient_ProcessLoop().
+ *
+ */
+typedef void ( * AzureIoTHubClientCloudToDeviceMessageCallback_t ) ( struct AzureIoTHubClientCloudToDeviceRequest * pxMessage,
+                                                             void * pvContext );
```

Comment:
alignment?

Reply:
weird looks fine on mine. I'll run uncrustify on it